src/Scrutor/ServiceCollectionExtensions.Decoration.cs

```diff
@@ -261,6 +276,27 @@ bool TryDecorate(Type[] typeArguments)
             return arguments.Aggregate(true, (result, args) => result && TryDecorate(args));
         }
 
+        private static bool TryDecorateOpenGeneric(this IServiceCollection services, Type serviceType, Func<object, IServiceProvider, object> decorator)
```

Comment:
Hmm... I'm not a big fan of the duplication here. I just merge #76, which made a change to the argument filter below:

```diff
- .Where(descriptor => descriptor.ServiceType.IsAssignableTo(serviceType))
+ .Where(descriptor => IsSameGenericType(descriptor.ServiceType, serviceType))
```

Ideally, this should be changed here as well. Are you able to find a way to share more of this code?

Reply:
Removed duplication by refining existed `TryDecorateOpenGeneric` method. 
Take a look, please.

Reply:
Refactored my implementation, simplified `TryDecorateOpenGeneric`.
Should look good now.

P.S. locally I have successful build, but AppVeyor fails with "unable to access 'https://github.com/khellang/Scrutor.git/': Could not resolve host: github.com"